packages/go_router/example/lib/stateful_shell_route.dart

```diff
@@ -0,0 +1,550 @@
+          title: Text('Details Screen - ${widget.label}'),
+        ),
+        body: _build(context),
+      );
+    } else {
+      return Container(
+        color: Theme.of(context).scaffoldBackgroundColor,
+        child: _build(context),
+      );
+    }
+  }
+
+  Widget _build(BuildContext context) {
+    return Center(
+      child: Column(
+        mainAxisSize: MainAxisSize.min,
+        children: <Widget>[
+          Text('Details for ${widget.label} - Counter: $_counter',
+              style: Theme.of(context).textTheme.titleLarge),
+          const Padding(padding: EdgeInsets.all(4)),
+          TextButton(
+            onPressed: () {
+              setState(() {
+                _counter++;
+              });
+            },
+            child: const Text('Increment counter'),
+          ),
+          const Padding(padding: EdgeInsets.all(8)),
+          if (widget.param != null)
+            Text('Parameter: ${widget.param!}',
+                style: Theme.of(context).textTheme.titleMedium),
+          const Padding(padding: EdgeInsets.all(8)),
+          if (widget.extra != null)
+            Text('Extra: ${widget.extra!}',
+                style: Theme.of(context).textTheme.titleMedium),
+          if (!widget.withScaffold) ...<Widget>[
+            const Padding(padding: EdgeInsets.all(16)),
+            TextButton(
+              onPressed: () {
+                GoRouter.of(context).pop();
+              },
+              child: const Text('< Back',
+                  style: TextStyle(fontWeight: FontWeight.bold, fontSize: 18)),
+            ),
+          ]
+        ],
+      ),
+    );
+  }
+}
+
+/// Widget for a modal screen.
+class ModalScreen extends StatelessWidget {
+  /// Creates a ModalScreen
+  const ModalScreen({Key? key}) : super(key: key);
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Modal'),
+      ),
+      body: Center(
+        child: Column(
+          mainAxisSize: MainAxisSize.min,
+          children: <Widget>[
+            Text('Modal screen', style: Theme.of(context).textTheme.titleLarge),
+            const Padding(padding: EdgeInsets.all(8)),
+            ElevatedButton(
+              onPressed: () {
+                GoRouter.of(context).go('/a');
+              },
+              child: const Text('Go to initial section'),
+            ),
+          ],
+        ),
+      ),
+    );
+  }
+}
+
+/// Builds a nested shell using a [TabBar] and [TabBarView].
+class TabbedRootScreen extends StatelessWidget {
+  /// Constructs a TabbedRootScreen
+  const TabbedRootScreen({Key? key}) : super(key: key);
+
+  Widget _child(StatefulShellBranchState branchState) => branchState.child;
+
+  @override
+  Widget build(BuildContext context) {
+    final StatefulShellRouteState shellState =
+        StatefulShellRouteState.of(context);
+    final List<Widget> children = shellState.branchStates.map(_child).toList();
```

Comment:
Good question, haven't tried that, but then again, that is not the idea or intent here. Instead, we're placing the navigators (children) in a custom container of our choosing (TabBarView), much like we would have if we'd built all this handling from scratch.

Reply:
I suggest we don't expose it for now unless there is an immediate use case people asked for. TabBarView is a pageView which does some non-trivial logic that you wouldn't want to know... It not only paint multiple child on screen at one time and it can also rearrange them in real time when you navigating between tab.

Even if we want to support this, let's separate that out into a different pr. I think that requires some thinking before we open the API up. WDYT?

Reply:
Well, there have been quite a few people wondering how to use StatefulShellRoute with for instance PageView, TabBarView and CupertinoTabScaffold etc. So I definitely think there is an immediate use case for this. And it's often not just about the container, but also about customising the animation when switching between Navigators/branches. 

So ideally, I think this should be included from the start, and exposing a children List seems pretty natural to me (in line with Stack/IndexedStack). Having said that, I recognise that this PR is already rather large, so if removing this field makes it substantially easier and faster to complete this PR, I could live with that.